File: frontend/src/types/mitigation.ts
import type { RiskLevel } from './risk'

export type MitigationStatus = 'pending' | 'pending_review' | 'approved' | 'rejected'
export type MeasureType = 'cleaning' | 'line_change'

export const mitigationLabels: Record<MitigationStatus, string> = {
  pending: '待处理',
  pending_review: '待复核',
  approved: '已缓解',
  rejected: '已拒绝',
}

export const measureTypeLabels: Record<MeasureType, string> = {
  cleaning: '清洗',
  line_change: '换线',
}

export interface MitigationMeasure {
  id: number
  route_id: number
  route_version: number
  allergen: string
  source_step_code: string
  target_step_code: string
  measure_type: MeasureType
  completed_on: string
  evidence_note: string
  raw_score: number
  risk_level: RiskLevel
  mitigation_status: MitigationStatus
  created_by: number
  reviewed_by?: number
  review_reason: string
  reviewed_at?: string
  created_at: string
  updated_at: string
}

export interface MitigationInfo {
  measure_id: number
  measure_type: MeasureType
  status: MitigationStatus
  completed_on: string
  evidence_note: string
}

File: backend/internal/dto/mitigation.go
package dto

type CreateMitigationRequest struct {
	RouteID        uint   `json:"route_id" validate:"required,min=1"`
	Allergen       string `json:"allergen" validate:"required,min=1,max=120"`
	SourceStepCode string `json:"source_step_code" validate:"required,min=2,max=64"`
	TargetStepCode string `json:"target_step_code" validate:"required,min=2,max=64,nefield=SourceStepCode"`
	MeasureType    string `json:"measure_type" validate:"required,oneof=cleaning line_change"`
	CompletedOn    string `json:"completed_on" validate:"required,len=10"`
	EvidenceNote   string `json:"evidence_note" validate:"required,min=4,max=1000"`
}

type ResubmitMitigationRequest struct {
	MeasureType  string `json:"measure_type" validate:"required,oneof=cleaning line_change"`
	CompletedOn  string `json:"completed_on" validate:"required,len=10"`
	EvidenceNote string `json:"evidence_note" validate:"required,min=4,max=1000"`
}

type ReviewMitigationRequest struct {
	Decision string `json:"decision" validate:"required,oneof=approved rejected"`
	Reason   string `json:"reason" validate:"required,min=4,max=1000"`
}

type MitigationQuery struct {
	Page     int    `form:"page" validate:"omitempty,min=1"`
	PageSize int    `form:"page_size" validate:"omitempty,min=1,max=100"`
	RouteID  uint   `form:"route_id"`
	Status   string `form:"status" validate:"omitempty,oneof=pending pending_review approved rejected"`
	Allergen string `form:"allergen" validate:"omitempty,max=120"`
}
